File: class/GearBox.py
###Boite de Vitesse###
#-Vitesse#
#-Ratio des Vitesse#

class GearBox:
    def __init__(self, name, gearRatio):
        self.gear = 0
        self.gearRatio = gearRatio

    def changeGear(self, gear):
        self.gear = gear

    def getGearRatio(self):
        if self.gear == "R":
            return self.gearRatio[0]
        if self.gear == "N":
            return self.gearRatio[1]
        if self.gear == "1":
            return self.gearRatio[2]
        if self.gear == "2":
            return self.gearRatio[3]
        if self.gear == "3":
            return self.gearRatio[4]
        if self.gear == "4":
            return self.gearRatio[5]
        if self.gear == "5":
            return self.gearRatio[6]

    def print(self):
        print(self.gear)
        print(self.gearRatio)

File: src/Vehicle/GearBox.cpp
#include "GearBox.hpp"

GearBox::GearBox(const char* Name, float* GearRatio){
  name = Name;
  gearRatio = GearRatio;
}

void GearBox::setProperties(const char* Name, float* GearRatio){
  name = Name;
  gearRatio = GearRatio;
}

void GearBox::setGearRatio(float* GearRatio){
  gearRatio = GearRatio;
}

void GearBox::changeGear(char Gear){ gear = Gear; }

float GearBox::getGearRatio(){
  switch (gear) {
    case 'R':
      return *gearRatio;
      break;
    case '1':
      return *(gearRatio + 2);
      break;
    case '2':
      return *(gearRatio + 3);
      break;
    case '3':
      return *(gearRatio + 4);
      break;
    case '4':
      return *(gearRatio + 5);
      break;
    case '5':
      return *(gearRatio + 6);
      break;
    default:
      return *(gearRatio + 1);
  }
}

void GearBox::print(){
  loger.log(gear);
}
